RPC Codecs for automatically capturing RPC messages

vaastav opened the pull request:
This PR adds custom RPC Codecs and convenience functions which use a govector logger to automatically capture all RPC Calls made by the client and the responses by the server.

bestchai commented:
This is great, thanks for working on this. A couple of comments:

- Can you add a GoVector log that is generated for the Server-Client example to the repo (and maybe even a png of the ShiViz visualization for this log). This way it the example will be clear end-to-end.

- The RPC example uses an HTTP-based connection. Any particular reason for this? If this is required, then I would document/note this. If this is not required, then maybe do away with the HTTP altogether? I would rather it was plain RPC (at least in 416 that's what most of our base assignments use).

vaastav commented:
The HTTP-based connection was just a remnant from some of my testing. I removed it.

I also added the log files generated, shiviz log as well as a png of the shiviz output.

bestchai commented:
Looks good, thanks!
